### PR TITLE
INSTUI-3890 simplify Docs build pipeline

### PR DESCRIPTION
**Pull request by matyasf**
Make typings better, remove some dead code.

TEST PLAN:
Compare the docs pages (especially the props part) with the current prod one and check if nothing is missing/changed

### CODE REVIEW

File: packages/__docs__/buildScripts/DataTypes.mts [hunk 1]
```diff
@@ -85,77 +92,61 @@ type JSDocFunctionReturns = {
     names: string[]
   }
 }
-// TODO remove these types, now we can get them directly from react-docgen
+// TODO these are from React-docgen Documentation.d.ts,
+// remove when react-docgen exports them
```
Comment (matyasf) [inline packages/__docs__/buildScripts/DataTypes.mts:96]: I've opened a ticket for them https://github.com/reactjs/react-docgen/issues/864

Comment (matyasf) [reply packages/__docs__/buildScripts/DataTypes.mts:96]: this code is just a refresh (copypaste) of their type definitions

File: packages/__docs__/buildScripts/utils/getJSDoc.mts [hunk 2]
```diff
@@ -55,18 +56,6 @@ export function getJSDoc(source: Buffer, error: (err: Error) => void) {
     }
     doc = {
       ...module,
-      sections: sections
```
Comment (matyasf) [inline packages/__docs__/buildScripts/utils/getJSDoc.mts:58]: this part was never used by the docs

File: packages/__docs__/src/Document/index.tsx [hunk 3]
```diff
@@ -368,31 +361,6 @@ import { ${importName} } from '${esPath}'
         )
     }
 
-    let sections
-
-    if (doc.sections) {
```
Comment (matyasf) [inline packages/__docs__/src/Document/index.tsx:373]: this was always empty

**Review by HerrTopi — APPROVED** [review]

**Pull request merged**